aspnet/mvc/models/validation.rst

```diff
@@ -41,172 +41,155 @@ Simply reading through the model reveals the rules about data for this app, maki
-	      // custom validation code
-          return false;
-      }, '');
+Unobtrusive validation uses the data in the ``data-`` attributes to display error messages. However, jQuery doesn't know about rules or messages until you add them to jQuery's ``validator`` object. This is shown in the example below that adds a method named ``classicmovie`` containing custom client validation code to the jQuery ``validator`` object.
 
-      jQuery.validator.unobtrusive.adapters.add('classicmovie',
-      function (options) {
-          options.rules['classicmovie'] = {};
-          options.messages['classicmovie'] = options.message;
-      });
-  }(jQuery));
+.. literalinclude:: validation/sample/Views/Movies/Create.cshtml
+   :language: javascript
+   :lines: 71-93
+   :dedent: 4
 
-Now jQuery has the information to execute the custom JavaScript validation as well as the error message to display if that validation code returns false. 
+Now jQuery has the information to execute the custom JavaScript validation as well as the error message to display if that validation code returns false.
 
 Remote validation
 -----------------
 
 Remote validation is a great feature to use when you need to validate data on the client against data on the server. For example, your app may need to verify whether an email or user name is already in use, and it must query a large amount of data to do so. Downloading large sets of data for validating one or a few fields consumes too many resources. It may also expose sensitive information. An alternative is to make a round-trip request to validate a field.
 
-You can implement remote validation in a two step process. First, you must annotate your model with the ``[Remote]`` attribute. The ``[Remote]`` attribute accepts multiple overloads you can use to direct client side JavaScript to the appropriate code to call. The example points to the ``VerifyEmail`` action method of the ``Users`` controller. 
+You can implement remote validation in a two step process. First, you must annotate your model with the ``[Remote]`` attribute. The ``[Remote]`` attribute accepts multiple overloads you can use to direct client side JavaScript to the appropriate code to call. The example points to the ``VerifyEmail`` action method of the ``Users`` controller.
 
 .. literalinclude:: validation/sample/User.cs
- :language: c#
- :lines: 6-24
- 
+   :language: c#
+   :lines: 5-9
+   :dedent: 4
+
 The second step is putting the validation code in the corresponding action method as defined in the ``[Remote]`` attribute. It returns a ``JsonResult`` that the client side can use to proceed or pause and display an error if needed.
- 
+
 .. literalinclude:: validation/sample/UsersController.cs
- :language: none
- :lines: 6-23
- 
-Now when users enter an email, JavaScript in the view makes a remote call to see if that email has been taken, and if so, then displays the error message. Otherwise, the user can submit the form as usual.  
+   :language: c#
```

Comment:
Looks like the C# lexer is failing to parse this one. To work around this replace `c#` with `none`